packages/datadog-instrumentations/src/mysql2.js

```diff
@@ -28,6 +31,76 @@ addHook({ name: 'mysql2', file: 'lib/connection.js', versions: ['>=1'] }, Connec
     return asyncResource.bind(addCommand, this).apply(this, arguments)
   })
 
+  shimmer.wrap(Connection.prototype, 'query', query => function (sql, values, cb) {
+    if (!startOuterQueryCh.hasSubscribers) return query.apply(this, arguments)
+
+    const sqlIsString = typeof sql === 'string'
+
+    const sqlString = sqlIsString ? sql : sql?.sql
```

Comment:
In other parts of this PR, this check is done differently. I think it would be worthwhile to homogenize all of them.

[Check 1](https://github.com/DataDog/dd-trace-js/pull/4712/files#diff-30989a4234a7039a68994bf4e35c149c3ef115c24b65682c793ce6301467168bR79)
[Check 2](https://github.com/DataDog/dd-trace-js/pull/4712/files#diff-30989a4234a7039a68994bf4e35c149c3ef115c24b65682c793ce6301467168bR163)
[Check 3](https://github.com/DataDog/dd-trace-js/pull/4712/files#diff-30989a4234a7039a68994bf4e35c149c3ef115c24b65682c793ce6301467168bR230)
[Check 4](https://github.com/DataDog/dd-trace-js/pull/4712/files#diff-30989a4234a7039a68994bf4e35c149c3ef115c24b65682c793ce6301467168bR264)